test/E2ETest/TestHelpers.cs

```diff
@@ -33,5 +43,91 @@ public static string GetSolutionRootDirectory(string solution)
 
             throw new Exception($"Solution file {solution}.sln could not be found in {applicationBasePath} or its parent directories.");
         }
+
+        public static async Task StartHostAndWaitForReplicasToStart(TyeHost host)
+        {
+            var startedTask = new TaskCompletionSource<bool>();
+            var alreadyStarted = 0;
+            var totalReplicas = host.Application.Services.Sum(s => s.Value.Description.Replicas);
```

Comment:
It seems like there's a logic error here - shouldn't the success condition be `totalReplicas - alreadStarted == 0`?

Reply:
It's counting up to totalReplicas and these description.replicas is the desired state, it's correct.